code/controllers/subsystem/gravity.dm

```diff
@@ -0,0 +1,173 @@
+var/datum/controller/subsystem/gravity/SSgravity
+
+/datum/controller/subsystem/gravity
+	name = "Gravity"
+	priority = 75
+	wait = 1
+	init_order = -100
+	flags = SS_KEEP_TIMING | SS_BACKGROUND
+
+	var/mob_base_gravity_slip_chance = 10
+	var/mob_base_gravity_fall_chance = 5
+	var/mob_handhold_gravity_slip_chance = 10
+	var/mob_handhold_gravity_fall_chance = 3
+	var/mob_gravity_strength_slip_mod = 3
+	var/mob_gravity_strength_fall_mod = 1.5
+	var/mob_slip_chance = 70
+	var/legacy_gravity = FALSE
+	var/list/currentrun = list()
+	var/list/currentrun_manual = list()
+	var/recalculation_cost = 0
+	var/do_purge = FALSE
+	var/purge_interval = 600
+	var/purge_tick = 0
+	var/purging = FALSE
+	var/error_no_atom
+	var/list/error_mismatched_area
+	var/list/error_mismatched_turf
+	var/list/error_no_area
+	var/list/error_no_turf
+	var/list/gravgens
+	var/list/purging_atoms
+	var/list/atoms_forced_gravity_processing
+	var/static/list/area_blacklist_typecache = typecacheof(list(/area/lavaland, /area/mine, /area/centcom))
+	var/inited_atoms = 0
+	var/inited_areas = 0
+	var/init_state = 0
+
+/datum/controller/subsystem/gravity/New()
+	NEW_SS_GLOBAL(SSgravity)
+
+/datum/controller/subsystem/gravity/proc/init_lists(full = FALSE)
+	gravgens = list()
+	purging_atoms = list()
+	atoms_forced_gravity_processing = list()
+	if(full)
+		error_no_turf = list()
+		error_no_area = list()
+		error_mismatched_turf = list()
+		error_mismatched_area = list()
+
+/datum/controller/subsystem/gravity/Initialize()
+	init_state = 1
+	init_lists(TRUE)
+	inited_atoms = 0
+	for(var/atom/movable/A in world)
+		A.init_gravity()
+		inited_atoms++
+		CHECK_TICK
+	for(var/area/A in world)
+		A.init_gravity()
```

Comment:
I don't believe adding full subsystem loop just in case master controller failure is good. The cost on standard run is too high. Unless i misunderstand this.